improver/regrid/landsea.py

```diff
@@ -271,46 +272,67 @@ def correct_where_input_true(self, selector_val: int) -> None:
                 Intended to be 1 for filling land points near the coast
                 and 0 for filling sea points near the coast.
         """
-        # Find all points on output grid matching selector_val
-        use_points = np.where(self.input_land.data == selector_val)
-
-        # If there are no matching points on the input grid, no alteration can
-        # be made. This tests the size of the y-coordinate of use_points.
-        if use_points[0].size == 0:
-            return
-
-        # Get shape of output grid
-        ynum, xnum = self.output_land.shape
-
-        # Using only these points, extrapolate to fill domain using nearest
-        # neighbour. This will generate a grid where the non-selector_val
-        # points are filled with the nearest value in the same mask
-        # classification.
-        (y_points, x_points) = np.mgrid[0:ynum, 0:xnum]
-        selector_data = griddata(
-            use_points,
-            self.nearest_cube.data[use_points],
-            (y_points, x_points),
-            method="nearest",
-        )
-
-        # Identify nearby points on regridded input_land that match the
-        # selector_value
-        if selector_val > 0.5:
-            thresholder = BasicThreshold(0.5)
-        else:
-            thresholder = BasicThreshold(0.5, comparison_operator="<=")
-        in_vicinity = self.vicinity(thresholder(self.input_land))
-
-        # Identify those points sourced from the opposite mask that are
-        # close to a source point of the correct mask
-        mismatch_points = np.logical_and(
-            np.logical_and(
-                self.output_land.data == selector_val,
-                self.input_land.data != selector_val,
-            ),
-            in_vicinity.data > 0.5,
-        )
+        if selector_val not in self._cache_nn:
+            # Find all points on output grid matching selector_val
+            use_points = np.where(self.input_land.data == selector_val)
+
+            # If there are no matching points on the input grid, no alteration can
+            # be made. This tests the size of the y-coordinate of use_points.
+            if use_points[0].size == 0:
+                return None, None
+
+            # Get shape of output grid
+            ynum, xnum = self.output_land.shape
+
+            # # Using only these points, extrapolate to fill domain using nearest
+            # # neighbour. This will generate a grid where the non-selector_val
+            # # points are filled with the nearest value in the same mask
+            # # classification.
+            (y_points, x_points) = np.mgrid[0:ynum, 0:xnum]
+
+            no_use_points = np.where(self.input_land.data != selector_val)
+            search_y = y_points[use_points].ravel()
```

Comment:
I think I might be missing something here, could you skip the `y_points[use_points]` and just use `use_points[0]`

Reply:
I think you are correct, but I have changed to defining a mask for use_points and no_use_points so that I can invert one to get the other so I'll have to leave as it is now.

Reply:
Hmm I'll change them back to np.where and avoid defining y_points etc. as you say.